packages/node_modules/@node-red/nodes/core/sequence/19-batch.js

```diff
@@ -229,10 +246,17 @@ module.exports = function(RED) {
                 return x.topic;
             });
             node.pending = {};
-            this.on("input", function(msg) {
-                concat_msg(node, msg);
+            this.on("input", function(msg,send,done) {
+                concat_msg(node, msg, send, done);
             });
             this.on("close", function() {
+                Object.values(node.pending).forEach(p_topic => {
+                    p_topic.groups.forEach(group => {
+                        group.msgs.forEach(msg => {
+                            msg.done("premature close");
```

Comment:
Is this really an error? In other nodes the close function calls `done()` with no error even though the message will be discarded.

We really need to be consistent.

Reply:
I forgot to comment to this.  I removed all occurrences of `"premature close"` argument in batch node on commit c23e71b.